### PR TITLE
feat: Drafts page implementation

### PR DESCRIPTION
**Pull request by meaghanfitzgerald**
### Related Ticket

https://eyblockchain.atlassian.net/browse/BIPT-42

### Description

Created a lazy implementation using a similar schema to the currently existing "All" page. I submit this PR so I can unblock the development of other features (via api implementation) but with full intentions to refactor most of front-end code into more concise, reusable components in the future. 

### QA Instructions

-[ ] checkout to `meag/draft-page-refactor`
-[ ] npm run dev
-[ ] toggle between the `All` and `Drafts` tabs to see the filtered vs. non-filtered pages

### Checklist

- [x] I have reviewed the code changes myself
- [x] I have removed unused code (e.g., `console.logs`, commented out blocks, etc.)
- [x] I have made all necessary changes to the documentation.
- [x] Any dependent changes have been merged and published.
- [x] Checked my PR title name is correct.

### CODE REVIEW

File: pages/api/drafts/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { PrismaClient, ProposalStatus } from "@prisma/client";
+import { withAuth } from "lib/auth";
+
+const prisma = new PrismaClient();
+
+export default withAuth(async (req, res, session) => {
+  const { method, body } = req;
```
Comment (nagrawal3) [inline pages/api/drafts/index.ts:7]:
```suggestion
  const { method } = req;
```

File: pages/api/drafts/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  switch (method) {
+    case "GET":
+      const drafts = await prisma.proposal.findMany({
```
Comment (nagrawal3) [inline pages/api/drafts/index.ts:11]: do we have error handling in place? what happens where db query throws error?

File: pages/api/drafts/index.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { PrismaClient, ProposalStatus } from "@prisma/client";
+import { withAuth } from "lib/auth";
+
+const prisma = new PrismaClient();
+
+export default withAuth(async (req, res, session) => {
+  const { method, body } = req;
+
+  switch (method) {
+    case "GET":
```
Comment (nagrawal3) [inline pages/api/drafts/index.ts:10]: maybe add a default case as well which returns 404 error

**Review by nagrawal3 — CHANGES_REQUESTED** [review]

File: pages/api/drafts/index.ts (hunk 3 above)
Comment (aj-may) [reply pages/api/drafts/index.ts:10]: or `405 Method Not Allowed`

File: pages/api/drafts/index.ts (hunk 1 above)
Comment (meaghanfitzgerald) [reply pages/api/drafts/index.ts]: resolved

File: pages/api/drafts/index.ts (hunk 3 above)
Comment (meaghanfitzgerald) [reply pages/api/drafts/index.ts:10]: had trouble implementing a default case without a valid variable, implemented error handling for the get method. In a future PR i may remove the `draft` endpoint and just write all the gets through the `proposal` endpoint.

File: package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "@chakra-ui/react": "^2.2.8",
     "@emotion/react": "^11.10.0",
     "@emotion/styled": "^11.10.0",
-    "@prisma/client": "^4.2.1",
+    "@prisma/client": "^4.4.0",
```
Comment (nagrawal3) [inline package.json:15]: when you are bumping version in `package.json` file, there should be changes in `package-lock.json` file. Can you please commit that as well

File: pages/api/proposals/index.ts [hunk 5]
```diff
@@ -9,6 +9,12 @@ export default withAuth(async (req, res, session) => {
   switch (method) {
     case "GET":
       const proposals = await prisma.proposal.findMany();
+
+      if (!proposals || proposals.length <= 0)
+        return res.status(404).json({
+          error: "No proposals found",
+        });
+
       return res.json(proposals);
```
Comment (nagrawal3) [inline pages/api/proposals/index.ts:18]:
```suggestion
      try {
        const proposals = await prisma.proposal.findMany();
        return res.json(proposals);
      } catch (err) {
        return res.status(500).json({
          error: "Error getting proposals",
        });
      }
```

Comment (nagrawal3) [reply pages/api/proposals/index.ts:18]: similar changes to other route as well

File: pages/api/drafts/index.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        where: { status: "DRAFT" },
+      });
+
+      if (!drafts || drafts.length <= 0)
+        return res.status(404).json({
+          error: "Failed to return any drafts",
+        });
+
+      return res.json(drafts);
+  }
```
Comment (nagrawal3) [inline pages/api/drafts/index.ts:21]:
```suggestion
  switch (method) {
    case "GET":
      try {
        const drafts = await prisma.proposal.findMany({
          where: { status: "DRAFT" },
        });
        return res.json(drafts);
      } catch(err) {
        return res.status(500).json({
          error: "Failed to get proposals",
        });
      }
      
    case default:
      return res.status(404).json({
        error: "No such route present",
      });
  }
```

**Review by nagrawal3 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (meaghanfitzgerald) [reply package.json:15]: resolved

File: pages/api/drafts/index.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      } catch(err) {
+        return res.status(500).json({
+          error: "Failed to get proposals",
+        });
+      }
+
+    default:
+      return res.status(404).json({
+        error: "No such route present",
+      });
```
Comment (nagrawal3) [inline pages/api/drafts/index.ts:25]:
```suggestion
      return res.status(405).json({
        error: "Method Not Allowed",
      });
```

**Review by nagrawal3 — APPROVED** [review]

**Pull request merged**